tests/test_position_transfers.py

```diff
@@ -0,0 +1,601 @@
+"""
+Tests for the DeriveClient transfer_position and transfer_positions methods.
+"""
+
+import pytest
+
+from derive_client.data_types import InstrumentType, TransferPosition
+
+PM_SUBACCOUNT_ID = 31049
+SM_SUBACCOUNT_ID = 30769
+TARGET_SUBACCOUNT_ID = 137404
+
+# Test instrument parameters
+TEST_INSTRUMENTS = [
+    ("ETH-PERP", InstrumentType.PERP, 2500.0, 0.1),
+    ("BTC-PERP", InstrumentType.PERP, 45000.0, 0.01),
+]
+
+# Position transfer amounts for testing
+TRANSFER_AMOUNTS = [0.1, 0.01, 0.5]
+
+
+def get_position_amount_for_test(derive_client, instrument_name, subaccount_id):
+    """Helper function to get position amount for testing, with fallback to mock data."""
+    try:
+        return derive_client.get_position_amount(instrument_name, subaccount_id)
+    except (ValueError, Exception):
+        pass  # If no position found or API call fails, use mock data
+
+    # Return mock position amount if no real position found
+    return 1.0
+
+
+@pytest.mark.parametrize(
+    "instrument_name,instrument_type,limit_price,amount",
+    TEST_INSTRUMENTS,
+)
+def test_transfer_position_basic(derive_client, instrument_name, instrument_type, limit_price, amount):
+    """Test basic transfer_position functionality."""
+    # Get available subaccounts
+    subaccounts = derive_client.fetch_subaccounts()
+    subaccount_ids = subaccounts['subaccount_ids']
+
+    # Use first two subaccounts for transfer
+    if len(subaccount_ids) < 2:
+        pytest.skip("Need at least 2 subaccounts for position transfer test")
+
+    from_subaccount_id = subaccount_ids[0]
+    to_subaccount_id = subaccount_ids[1]
+
+    # Get position amount for testing (with fallback to mock data)
+    position_amount = get_position_amount_for_test(derive_client, instrument_name, from_subaccount_id)
+    result = derive_client.transfer_position(
+        instrument_name=instrument_name,
+        amount=amount,
+        limit_price=limit_price,
+        from_subaccount_id=from_subaccount_id,
+        to_subaccount_id=to_subaccount_id,
+        position_amount=position_amount,
+    )
+
+    assert result is not None
+    assert hasattr(result, 'transaction_id')
+    assert hasattr(result, 'status')
+    assert hasattr(result, 'tx_hash')
+
+
+@pytest.mark.parametrize(
+    "from_subaccount,to_subaccount",
+    [
+        (SM_SUBACCOUNT_ID, PM_SUBACCOUNT_ID),
+        (PM_SUBACCOUNT_ID, SM_SUBACCOUNT_ID),
+    ],
+)
+def test_transfer_position_between_specific_subaccounts(derive_client, from_subaccount, to_subaccount):
+    """Test transfer_position between specific subaccount types."""
+    instrument_name = "ETH-PERP"
+    amount = 0.1
```

Comment:
Define these as constants so they can be reused rather than magic vars through the tests